doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,118 @@
+
+```cpp
+  /* 
+  *  Option 1. Init a new client with the same HTTP transport adapter.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions optionsA;
+  optionsA.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientA = BlobServiceClient(url, credential, options);
+  // The second client.
+  BlobClientOptions optionsB;
+  optionsB.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientB = BlobServiceClient(url, credential, options);
+
+  /* 
+  *  Option 2. Create new client from a parent client.
+  */
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>();
+  BlobClientOptions optionsA;
+  optionsA.TransportPolicyOptions.Transport = curlTransportAdapter;
+  auto storageClientA = BlobServiceClient(url, credential, options);
+  // Create clientB from clientA.
+  auto clientB = storageClientA.GetBlobContainerClient(containerName)
+```
+
+Note that the second option could be not supported for some Azure SDK clients if it doesn't make sense for the service behavior (like Key Vault client).
+
+The Azure SDK for C++ client `will hold an internal copy of the options` passed in the arguments. This means that the options object used to init the client can be disposed. However, since the HTTP transport adapter is a shared ptr, it won't be disposed until any client using it is disposed first.
+
+## HTTP Transport Adapter Options
+The libcurl and WinHTTP transport adapters can also be init with specif options that control features from the HTTP stack underneath it.
+For example, you can set a PROXY or override the default CA path for libcurl. See next example.
+```cpp
+  /*
+  * Customize libcurl http transport adapter.
+  */
+
+  // Create HTTP transport adapter options and set options.
+  Azure::Core::Http::CurlTransportOptions curlTransportOption;
+  curlTransportOption.Proxy = "https://my.proxy.com";
+  // Create HTTP transport adapter with options.
+  auto curlTransportAdapter = std::make_shared<Azure::Core::Http::CurlTransport>(curlTransportOption);
+
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = curlTransportAdapter; 
+  auto storageClient = BlobServiceClient(url, credential, options);
+```
+
+The HTTP transport adapter are specific for each adapter and are optional. This means that if you are creating your own HTTP transport adapter, you don't need to create options for it if you don't want to.
+
+
+## Building a custom HTTP Transport Adapter
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
+
+You can see [CMake options](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#cmake-build-options) to learn about all the supported options and what is the specific option required to be set when building your own HTTP transport adapter.
+
+Follow next steps to implement your own HTTP Transport Adapter.
+
```

Comment:
There seems to be just one step. Remove the numbered list?

Reply:
Added the missing steps.
Somehow I forgot to continue and finish with all the steps I was hoping to add initially. I think I got lost on step 1 and forgot about it hahahah